org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/text/correction/proposals/InitializeFinalFieldAtConstructorDescription.java

```diff
@@ -0,0 +1,37 @@
+/*******************************************************************************
```

Comment:
Hi @zelenyhleb Thanks for contributing.  A new class is not warranted for this.  Just add a conditional using the updateType in the super constructor call in InitializeFinalFieldProposal.

Reply:
Hi @jjohnstn. Thank you for your review! I did this initially to improve readability a bit, because with conditionals constructor looked overloaded for me. Fixed this.